src/App.tsx

```diff
@@ -29,20 +30,23 @@ function App() {
 
   return (
     <LoginContext.Provider value={{ user, setUser }}>
-      <div className="flex min-h-screen flex-col justify-between">
-        {user ? <Menu /> : null}
-        <Router>
+      <div className="flex min-h-screen flex-col justify-between">      
+        <Router> 
+        {user ? <Menu /> : null} 
           <Routes>
             <Route
               path="/"
               element={<LoginPage />}
               errorElement={<Error404 />}
             />
-            <Route path="/events" element={<Homepage />}>
-              <Route path=":alertID" element={<Alert message="hi" />} />
-              {/* alertID needs to be updated to display events */}
-            </Route>
-            <Route path="*" element={<Error404 />} />
+            <Route element={<RequireAuth />}>
```

Comment:
Really really like this usage of outlet here!!

Reply:
Chan Mi is gassing u up suli

Reply:
LETS GOO THANK YOU